src/Assignments/EmailProfessor.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import { postEmailProfessor } from '../APIInterfaces/BackendAPI/Requests/CourseRequests';
+import { ProblemObject, TopicObject } from '../Courses/CourseInterfaces';
+import { useCourseContext } from '../Courses/CourseProvider';
+import { Alert, Button, Form, FormControl, FormGroup, FormLabel, Modal, ModalBody, ModalFooter, ModalTitle } from 'react-bootstrap';
+import ModalHeader from 'react-bootstrap/ModalHeader';
+import useAlertState from '../Hooks/useAlertState';
+
+interface EmailProfessorProps {
+    topic: TopicObject;
+    problem: ProblemObject;
+}
+
+/**
+ * This is a button-and-model pair.
+ */
+export const EmailProfessor: React.FC<EmailProfessorProps> = ({problem}) => {
+    const {course} = useCourseContext();
+    const [content, setContent] = useState<string>('');
+    const [show, setShow] = useState<boolean>(false);
+    const [{message: sendEmailRespMsg, variant: sendEmailRespAlertType}, setSendEmailRespMsg] = useAlertState();
+
+    const onClick = async () => {
+        try {
+            const res = await postEmailProfessor({
+                courseId: course.id,
+                content: content,
+                question: {
+                    id: problem.id,
+                }
+            });
+            setSendEmailRespMsg({message: res.data.message ?? 'Email sent.', variant: 'success'});
+        } catch (e) {
+            console.error(e);
```

Comment:
👿 👿 👿 👿 👿

Reply:
This isn't a nit anymore since we have the logger implemented, especially since it is an error this won't bubble up
If we think it can validly fail then we should change it to a warn